main/src/ca/uwaterloo/flix/language/phase/jvm/GenContext.scala

```diff
@@ -17,14 +17,84 @@
 package ca.uwaterloo.flix.language.phase.jvm
 
 import ca.uwaterloo.flix.api.Flix
-import ca.uwaterloo.flix.language.ast.ExecutableAst.Root
-import ca.uwaterloo.flix.language.ast.Type
+import ca.uwaterloo.flix.language.ast.{ExecutableAst, Symbol}
+import org.objectweb.asm.ClassWriter
+import org.objectweb.asm.Opcodes._
 
 object GenContext {
 
   // TODO: Documentation and signature
-  def gen(ts: Set[Type], root: Root)(implicit flix: Flix): Map[JvmName, JvmClass] = {
-    Map.empty // TODO
+  def gen(defns: Map[Symbol.DefnSym, ExecutableAst.Def])(implicit flix: Flix): Map[JvmName, JvmClass] = {
+
+    // Class visitor
+    val visitor = new ClassWriter(ClassWriter.COMPUTE_FRAMES){
+      override def getCommonSuperClass(tpe1: String, tpe2: String) : String = {
+        JvmType.Obj.name.toInternalName
+      }
+    }
+
+    // Class header
+    visitor.visit(JvmOps.JavaVersion, ACC_PUBLIC + ACC_FINAL, JvmType.Context.name.toInternalName, null,
+      JvmType.Obj.name.toInternalName, null)
+
+    // Namespaces
+    val namespaces = defns.keys.map(_.prefix).toSet
+
+    // Adding continuation field
+    JvmOps.compileField(visitor, "continuation", JvmType.Obj.toDescriptor, isStatic = false, isPrivate = false)
+
+    // Adding field for each namespace
+    namespaces.foreach{ namespace =>
+      val namespaceRef = JvmOps.getNamespaceType(namespace)
+      JvmOps.compileField(visitor, namespaceRef.name.name, namespaceRef.toDescriptor, isStatic = false, isPrivate = false)
+    }
+
+
+    // Add the constructor
+    compileContextConstructor(visitor, namespaces)
+
+    visitor.visitEnd()
+    Map(JvmType.Context.name -> JvmClass(JvmType.Context.name, visitor.toByteArray))
+  }
+
+  /**
+    * Add the constructor for the class which initializes each field
+    */
+  private def compileContextConstructor(visitor: ClassWriter, namespaces: Set[List[String]]): Unit = {
+
+    // Method header
+    val constructor = visitor.visitMethod(ACC_PUBLIC + ACC_FINAL, "<init>", "()V", null, null)
+    constructor.visitCode()
+
+    constructor.visitCode()
+    constructor.visitVarInsn(ALOAD, 0)
+
+    // Call the super (java.lang.Object) constructor
+    constructor.visitMethodInsn(INVOKESPECIAL, JvmType.Obj.name.toInternalName, "<init>",
+      "()V", false)
+
+    // Initializing each field
+    namespaces.foreach{ namespace =>
+
+      // JvmType of the namespace
+      val namespaceRef = JvmOps.getNamespaceType(namespace)
```

Comment:
Okay, so I understand better now, a namespace type is the type of the namespace class. Conceptually this is a bit problematic because we know have JvmTypes that do not correspond to any flix type. I will think more about it.